connect-inject/container_init_test.go

```diff
@@ -1514,3 +1517,28 @@ func TestHandlerContainerInit_Resources(t *testing.T) {
 		},
 	}, container.Resources)
 }
+
+func TestHandlerContainerInit_MismatchedServiceNameServiceAccountName(t *testing.T) {
+	require := require.New(t)
+	h := Handler{
+		AuthMethod: "auth-method",
+	}
+	pod := &corev1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Annotations: map[string]string{
+				annotationService: "foo",
+			},
+		},
+		Spec: corev1.PodSpec{
+			Containers: []corev1.Container{
+				{
+					Name: "serviceName",
+				},
+			},
+			ServiceAccountName: "notServiceName",
+		},
+	}
+
+	_, err := h.containerInit(pod, k8sNamespace)
+	require.EqualError(err, `serviceAccountName "notServiceName" does not match service name "foo"`)
+}
```

Comment:
Can you add another test for when there is no annotation and we're using the first pod name and also for when ACLs are not enabled and so the names can be mismatched with no error.

Reply:
I thought of writing that test but between https://github.com/hashicorp/consul-k8s/blob/7e9d6501f8431f9b1b7997b4a23341a4435758f7/connect-inject/container_init.go#L79:L83 and https://github.com/hashicorp/consul-k8s/blob/7e9d6501f8431f9b1b7997b4a23341a4435758f7/connect-inject/handler.go#L425:L434, the annotation for `serviceName` is always set and it will panic if the annotation is not.

Reply:
Gotcha, makes sense!

Reply:
Oh, what about a test for when ACLs are not enabled? i.e. `Handler.AuthMethod == ""`.

Reply:
there are other tests up top that would fail and hence i didn’t write one explicitly. i can add one though.

Reply:
Yeah I think an explicit one makes sense here. I looked at the other tests and they don't seem targeted enough to this use-case imo.

Reply:
Done 💥